### ISSUE TITLE
Categories should be re-fetched again after updated

### ISSUE THREAD

**Issue by neslihanturan** [commons-app/apps-android-commons#3890]
**Summary:** 
Currently we have the function to edit categories from the app: #3851 . But unfortunately, we have a logic to fetch uploads only once when the contrib list is being loaded until we close the app. So, if the user changes the categories both from the app or from web interface, our contrib list remains same since it is fetched from db, not from servers. We should:
a- refetch all whenever an update is made (inefficient in terms of bandwidth)
b- fetch the specific field and update the db accordingly (most efficient in terms of bandwidth)
c- having a class to re-fetch the contribution from servers with all fields (so that we can use the same class to update descriptions or license along with categories). Then update the db with new contribution with contributionsRepository.updateContribution(Contribution contribution)  method.

Related classes to edit our local db are ContributionsPresenter, ContributionsListPresenter and ContributionsRepository.updateContribution(Contribution contribution)

**Comment by NotSoImportant** [comment]
yes, the app says something like "update success", but when looking at details it is not reflected there, which makes the user feel like: was it really updated?

**Comment by Ayan-10** [comment]
@neslihanturan @nicolas-raoul Can I work on this issue?

**Comment by nicolas-raoul** [comment]
I think the main problem has been fixed already?
I just tested: when I add a category within the app, it is immediatly visible.

Everything (including the filename and the image's binary itself) can be modified outside of the app, but for bandwidth reasons we can not update everything everytime, right? How about refreshing everything only if the user explicitely requests it, for instance with a swipe to the top?

**Comment by Ayan-10** [comment]
@nicolas-raoul Yes, I know the problem has fixed now. But It just showing the new categories. It doesn't refetch the categories from the server. The app has a short-term hack to show the new categories. You can see in the image that there is a TODO in the update categories. But yes it's showing the updated categories. But as @neslihanturan mentioned in point "c". We can have a class for getting description, categories, licenses e.t.c. from the server and fetch with the help of the one class when required. (Fetch the new categories with the help of the class)

Also, your idea of refreshing everything when user swipes to the top is great.

<img width="627" alt="2021-07-04" src="https://user-images.githubusercontent.com/71203077/124374468-0bcd5a80-dcb9-11eb-8c00-4229d273e2d4.png">

So what should I do? Should I keep the `updateCategories` logic as it is in the app and  refresh everything when user swipes to the top or should I follow the "c" point of @neslihanturan

**Comment by Ayan-10** [comment]
@nicolas-raoul Can you put your thoughts on this?

**Comment by nicolas-raoul** [comment]
The current behavior is not shocking to me.
How about keeping it, removing the TODO comment, and implementing a swipe gesture that does what Neslihan described in the "c" point?

**Comment by Ayan-10** [comment]
@nicolas-raoul Ok, I will do that. So when user swipes top the app should fetch all sections of the contribution item from the server. I think description, license, location, categories can be fetched together using `getWikiText` method. Is there any API for fetching Filename and image from the server for a contribution?
